testing/MLDB-1468-credentials-test.py

```diff
@@ -5,32 +5,118 @@
 #
 
 import unittest
+from os.path import isfile, join, expanduser
 
 mldb = mldb_wrapper.wrap(mldb) # noqa
 
 class CredentialTest(MldbUnitTest):
 
-    def test_creation_of_volatile_creds(self):
+    def test_creation_of_dummy_creds(self):
         # try something that should work
         # mldb.get asserts the result status_code is >= 200 and < 400
         resp = mldb.put("/v1/credentials/s3cred", {
             "store" : {
                 "resourceType" : "aws:s3",
                 "resource" : "s3://",
-                "role" : "",
-                "operation" : "",
-                "expiration" : "",
                 "credential" : {
-                    "provider" : "",
+                    "provider" : "Credentials collection",
                     "protocol" : "http",
                     "location" : "s3.amazonaws.com",
                     "id" : "this is my key",
-                    "secret" : "this is my secret",
-                    "validUntil" : "2016-01-01"
+                    "secret" : "this is my secret"
                 }
             }
         })
 
         mldb.log(resp)
 
+    def test_selection_of_creds(self):
+        # store a dummy credential for a specific path
+        resp = mldb.put("/v1/credentials/badcred", {
+            "store" : {
+                "resourceType" : "aws:s3",
+                "resource" : "s3://dummy",
+                "credential" : {
+                    "provider" : "Credentials collection",
+                    "protocol" : "http",
+                    "location" : "s3.amazonaws.com",
+                    "id" : "this is my key",
+                    "secret" : "this is my secret"
+                }
+            }
+        })
+
+        csv_conf = {
+            "type": "import.text",
+            "params": {
+                'dataFileUrl' : 's3://dummy/test.csv',
+                "outputDataset": {
+                    "id": "test"
+                },
+                "runOnCreation": True
+            }
+        }
+
+        # this is expected to pick the most specific but invalid credentials
+        with self.assertRaises(mldb_wrapper.ResponseException) as re:
+            resp = mldb.put("/v1/procedures/import", csv_conf)
+
+
+    @unittest.skipIf(not isfile(join(expanduser('~'), '.cloud_credentials')),
```

Comment:
Jenkins includes a .cloud_credentials file so all tests that require credentials to run including this one are run on every build.  Not sure how to implement this.  There are some tests that we want to skip (e.g those that are marked as manual because they are awaiting a fix) and some other that should not be skipped.  Any idea how we would do that?